packages/core/src/index.ts

```diff
@@ -306,7 +306,11 @@ export async function createBundleHandler(
     logger
   );
 
-  if (resolvedUserConfig.compilation?.watch || watchMode) {
```

Comment:
We allow users to configure the development mode to production

Reply:
OK, do you think just close it in `build` after watcher created or do you think to convey a new var to `createBundleHandler` to fix #1597?

Reply:
I found it `watchMode` will only be true in `watch` command, It seems to be ok to just use `watchMode` to judge whether create a watcher?